File: week 6/lecture/mario.py
def main():
    while True:
        height = int(input("Height: "))
        if height >= 1 and height <= 10000:
            break

    for i in range(height):
        for j in range(height):
            if i + j >= height - 1:
                print("#", end="")
            else:
                print(" ", end="")
        print(" ", end="")

        for k in range(0, i + 1):
            print("#", end="")
        print()


if __name__ == '__main__':
    main()

File: week 1/pset/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Get input from the user and keep asking in case the user enters an invalid input
    int input;
    do
    {
        input = get_int("Height: ");
    }
    while (input < 1 || input > 8); 

    for (int i = 0; i < input; i++)
    {
        for (int j = 0; j < input; j++)
        {
            // Check if the index is greater than input, so it right alingns the pyramid
            if (i + j >= input - 1) 
            {
                printf("#");
            }
            else
            {
                printf(" ");
            }
        }
        // Prints two spaces in the middle of the pyramid
        printf("  "); 

        // Draw the right half of the pyramid
        for (int k = 0; k <= i; k++) 
        {
            printf("#");
        }

        printf("\n");
    }
}